test/compress/yield.js

```diff
@@ -199,3 +197,21 @@ yield_as_ES5_property: {
     expect_exact: '"use strict";console.log({yield:42}.yield);'
     expect_stdout: "42"
 }
+
+issue_2689: {
+    options = {
+        collapse_vars: true,
+        unused: true,
+    }
+    input: {
+        function* y() {
+            var t = yield x();
+            return new t();
+        }
+    }
+    expect: {
+        function* y() {
+            return new (yield x());
```

Comment:
should be `expect_exact`

Reply:
`expect_exact` wouldn't parse the output back, which makes the comparison not as clear IMHO as this:
```
    Running test [issue_2689]
!!! Test matched expected result but cannot parse output
---INPUT---
{
    function* y() {
        var t = yield x();
        return new t();
    }
}
---OUTPUT---
function*y(){return new yield x()}
--REPARSE ERROR--
SyntaxError: Unexpected token: name (x)
```
... which feels like a stronger indicator that something is wrong.

Reply:
> `expect_exact` wouldn't parse the output back

I did not know that. It probably should - just to verify the output is parseable.

Reply:
Oops, mea culpa - `expect_exact` does eventually try and parse the output back. I just need to supply a matching (incorrect) string to trigger it:
```diff
diff --git a/lib/output.js b/lib/output.js
index 47d0767..2f8faf7 100644
--- a/lib/output.js
+++ b/lib/output.js
@@ -799,10 +799,6 @@ function OutputStream(options) {
         // a = yield 3
         if (p instanceof AST_Binary && p.operator !== "=")
             return true;
-        // (yield 1)()
-        // new (yield 1)()
-        if (p instanceof AST_Call && p.expression === this)
-            return true;
         // (yield 1) ? yield 2 : yield 3
         if (p instanceof AST_Conditional && p.condition === this)
             return true;
diff --git a/test/compress/yield.js b/test/compress/yield.js
index cb11147..4ef38cc 100644
--- a/test/compress/yield.js
+++ b/test/compress/yield.js
@@ -209,6 +209,7 @@ issue_2689: {
             return new t();
         }
     }
+    expect_exact: "function*y(){return new yield x()};"
     expect: {
         function* y() {
             return new (yield x());
```
```
    Running test [issue_2689]
!!! Test matched expected result but cannot parse output
---INPUT---
{
    function* y() {
        var t = yield x();
        return new t();
    }
}
---OUTPUT---
function*y(){return new yield x()}
--REPARSE ERROR--
SyntaxError: Unexpected token: name (x)
```
Will change the test back to `expect_exact` in a moment.